vispy/glsl/misc/spatial-filters.frag

```diff
@@ -4,23 +4,47 @@
 
 const float kernel_bias  = -0.234377;
 const float kernel_scale = 1.241974;
+const float kernel_size = 1024;
 uniform sampler2D u_kernel;
 
+float unpack(vec4 rgba) {
+    //return rgba.r;  // uncomment this for r32f debugging
+    rgba.rgba = rgba.abgr * 255;
+    float sign = 1.0 - step(128.0,rgba[0])*2.0;
+    float exponent = 2.0 * mod(rgba[0],128.0) + step(128.0,rgba[1]) - 127.0;
+    float mantissa = mod(rgba[1],128.0)*65536.0 + rgba[2]*256.0 + rgba[3] + float(0x800000);
+    return sign * exp2(exponent) * (mantissa * exp2(-23.0));
+}
+
+float unpack_interpolate(sampler2D kernel, vec2 uv) {
+    //return texture2D(kernel, uv).r; //uncomment this for r32f debug without interpolation
+    float kpixel = 1/kernel_size;
```

Comment:
So my error was thinking `shape` was related to the `kernel` when really it was related to the data to be interpolated. Makes sense